### ISSUE TITLE
#[macro_use] use path; produces unhelpful diagnostics

### ISSUE THREAD

**Issue by ExpHP** [rust-lang/rust#51665]
```rust

mod a {
    macro_rules! foo { () => {}; }
}

mod b {
    #[macro_use]
    use a;
    
    foo!();
}

fn main() {}
```

```
   Compiling playground v0.0.1 (file:///playground)
error: cannot find macro `foo!` in this scope
  --> src/main.rs:10:5
   |
10 |     foo!();
   |     ^^^
   |
   = help: have you added the `#[macro_use]` on the module/import?
```

The error message even seems to suggest that this is what you're supposed to write, by using the phrase "on the module/**import**"

**Comment by estebank** [comment]
Suggested output:

```
   Compiling playground v0.0.1 (file:///playground)
error: cannot find macro `foo!` in this scope
  --> src/main.rs:10:5
   |
10 |     foo!();
   |     ^^^ not found in this scope
   |
help: include the `foo!` macro from `mod a` in the scope:
 --> src/main.rs:1:0
  |
1 | #[macro_use] mod a {
  | ^^^^^^^^^^^^
```

**Comment by ExpHP** [comment]
That sounds good; once the code sample compiles, the attribute on the `use` does generate a warning, so it'll be good to drive the user to that point.

Mind also that the user's issue could potentially be that modules are listed in the wrong order.
